tutorials/review.adoc

```diff
@@ -0,0 +1,881 @@
+
+[source, shell]
+----
+metanorma document.adoc
+----
+With this command you trigger the metanorma toolchain to:
+
+* Read the AsciiDoc input
+* Convert it into XML
+* Check the XML against the document model (XML schema)
+* Create HTML, PDF, and DOC output
+
+[[exercise-3-2]]
+include::exercises.adoc[tag=exercise-3-2]
+
+
+Now that you are familiar with the simplest way of creating a Metanorma document, let's look at different ways to compile documents in the next lesson.
+>> Start next lesson
+
+== Publishing a Metanorma Document
+tag::[learningobjectives-4]
+
+Once you are happy with your document, the next step is to generate an output to send your SDO for publishing. Since Metanorma poses some validity criteria for your document, it is normal to face some errors. Don’t worry, we will go through some common errors and teach you how to resolve them. 
+In this lesson you will learn:
+
+* How to generate PDF, HTML, and Word output using the command line
+* How to troubleshoot common errors 
+
+end::[learningobjectives-4]
+
+>> Start next lesson
+
+=== Compiling a Metanorma Document
+
+To successfully compile a Metanorma document, the toolchain looks at:
+
+* Metadata: The title, document flavor, and document type.
+* Syntax: Is the entered AsciiDoc code correct?
+* Dependencies: Are all tools needed to compile the document installed?
+// Anything else?
+
+To compile a Metanorma document:
+
+. On the command line, go to the folder where the document you want to compile is located. 
+. Enter the following command: +
+[source, shell]
+----
+metanorma document.adoc
+----
+With this command you trigger the metanorma toolchain to:
+
+* Read the AsciiDoc input
+* Convert it into XML
+* Check the XML against the document model (XML schema)
+* Create HTML, PDF, and DOC output
+
+You can also manipulate the way Metanorma compiles a document by setting flags. Flags are appended to the build command, like this:  
+
+.Example of a build command with a flag
+[source, shell]
+----
+metanorma document.adoc -x html
+----
+
+You can use the following flags to manipulate the building process:
+
+* `-t`: Sets the flavor of the document +
+NOTE: Either define the flavor in the metadata using `mn-document-class` or use `-t flavor`
+* `-x`: Sets the output format(s) of the document 
+* `-o`: Enter a path that the output should be saved to. 
+
+To see the full list of possible build commands, open the metanorma help on the command line.
+
+[source, shell]
+----
+metanorma help compile
+----
+
+>> Start next lesson
+
+=== Troubleshooting 
+
+There can be many reasons why you can’t compile a Metanorma AsciiDoc document into the final output, but you can easily fix them when you know what typical errors occur and how to fix them. 
+Metanorma tells you what’s wrong in the terminal and stores all errors in an Error file `.err` where your original document is stored. 
```

Comment:
```suggestion
Metanorma tells you what’s wrong in the terminal and stores all errors in an Error file `.err` where your original document is stored. If execution has aborted before the XML content could be finalised, the XML file is still output to disk, suffixed with `.xml.abort` rather than `.xml`, and you can use it to make sense of error messages.
```